lib/origen/remote_manager.rb

```diff
@@ -243,6 +281,11 @@ def remotes
     #   * If multiple versions of the same remote are found the most
     #     recent one wins.
     def add_remote(new)
+      # Cannot have both a tag and a version defined for a remote
+      if ([:tag, :version] - new.keys).empty?
+        Origen.log.error('Cannot define both a tag and a version for a remote!')
```

Comment:
This should be:

~~~ruby
fail "Cannot..."
~~~

i.e.  better to show the user the stack trace if going to kill the execution